frame/im-online/src/benchmarking.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Balances pallet benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_system::RawOrigin;
+use frame_benchmarking::benchmarks;
+use sp_core::offchain::{OpaquePeerId, OpaqueMultiaddr};
+use sp_runtime::traits::{ValidateUnsigned, Zero};
+
+use crate::Module as ImOnline;
+
+const MAX_KEYS: u32 = 1000;
+
+pub fn create_heartbeat<T: Trait>(k: u32) ->
+	Result<(crate::Heartbeat<T::BlockNumber>, <T::AuthorityId as RuntimeAppPublic>::Signature), &'static str>
+{
+	let mut keys = Vec::new();
+	for _ in 0 .. k {
+		keys.push(T::AuthorityId::generate_pair(None));
+	}
+	Keys::<T>::put(keys);
+
+	let network_state = OpaqueNetworkState {
+		peer_id: OpaquePeerId::default(),
+		external_addresses: vec![OpaqueMultiaddr::new(vec![]); 10],
```

Comment:
Maybe check if `external_addresses` is reasonable, or make it a parameter (I think we are encoding/decoding it)

Reply:
Yeah, Should figure out what the right size is here. I will add some constants that we can tweak if needed.